### PR TITLE
Adding more C# examples

### PR DESCRIPTION
**Pull request by jbw3**
Most of the examples show how to use a .NET library from Python. However, there are not many that show how to use Python in C#, so I've added several.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -517,13 +517,74 @@ which may be called at startup without having acquired the GIL.
 
 When finished using Python APIs, managed code must call a corresponding
 `PythonEngine.ReleaseLock` to release the GIL and allow other threads
-to use Python.
+to use Python:
+
+```csharp
+IntPtr gilState = PythonEngine.AcquireLock();
```
Comment (filmor) [inline README.md]: We should not encourage this, the `Py.GIL` part should be enough.

File: README.md [hunk 2]
```diff
@@ -517,13 +517,74 @@ which may be called at startup without having acquired the GIL.
...
+
+// acquire the GIL before using the Python interpreter
+using (Py.GIL())
+{
+    // convert the Person object to a PyObject
+    PyObject pyPerson = person.ToPython();
+
+    // create a Python variable "person"
+    PyDict locals = new PyDict();
+    locals["person"] = pyPerson;
```
Comment (filmor) [inline README.md]: We have the `PyScope` interface to make this a bit easier.

**Comment by jbw3** [comment]
I have addressed the comments.

**Comment by filmor** [comment]
Great, thank you very much :)

**Pull request merged**